### PR TITLE
Fixes and Extended Support for SolidRun i.MX6 Devices

### PR DESCRIPTION
**Pull request by Artox**
1. Fix an issue where SPL fails to load u-boot.img from the first partition (tested on HummingBoard 2)

2. Add **all the DTBs** - especially all the SoM rev 1.5 variants

3. simplify boot-script a lot so that it uses all of U-Boots knowledge in place of custom device identification logic

### CODE REVIEW

**Comment by Artox** [comment]
The final commit should not be merged **yet** - it depends on a patch that only just landed in master and will tehrefore only be part of Linux 5.9. So once the wifi drivers backport jumps to 5.9 - the final patch can be applied.

File: target/linux/imx6/image/Makefile [hunk 1]
```diff
@@ -206,7 +206,10 @@ define Device/cubox-i
   UBOOT := mx6cuboxi
   BOOT_SCRIPT = bootscript-cubox
   DEVICE_NAME := cubox
-  DEVICE_PACKAGES := kmod-drm-imx kmod-drm-imx-hdmi kmod-usb-hid
+  DEVICE_PACKAGES := \
+	kmod-drm-imx kmod-drm-imx-hdmi \
+	kmod-brcmfmac brcmfmac-firmware-4329-sdio brcmfmac-firmware-4329-sdio-cubox-i brcmfmac-firmware-4330-sdio brcmfmac-firmware-4330-sdio-cubox-i \
+	kmod-wl18xx wl18xx-firmware wl18xx-firmware-cubox-i
```
Comment (adrianschmutzler) [inline target/linux/imx6/image/Makefile:212]: Consider wrapping these lines a little more systematically, i.e. just before 80 characters for all lines or similar.

Comment (Artox) [reply target/linux/imx6/image/Makefile:212]: well, I was tryi8ng to be systematic by having line by feature more or less
- display
- Broadcom wifi
- Ti wifi
However I will think about reorganizing. Any ideas?

File: package/kernel/mac80211/broadcom.mk [hunk 2]
```diff
@@ -451,6 +451,7 @@ define KernelPackage/brcmfmac/config
 		bool "Enable SDIO bus interface support"
 		default y if TARGET_bcm27xx
 		default y if TARGET_sunxi
+		default y if TARGET_imx6_DEVICE_cubox-i
```
Comment (adschm) [inline package/kernel/mac80211/broadcom.mk:454]: I don't think this kind of package selection will work per device, but only per target.

File: target/linux/imx6/image/Makefile (hunk 1 above)
Comment (adschm) [reply target/linux/imx6/image/Makefile:212]: Well, one line is longer than allowed anyway, either wrap each line when full so no line is longer than 74 chars or just have one package per line.

**Comment by ynezz** [comment]
What's the status of this? If you're not interested anymore then please close this PR.

**Comment by Artox** [comment]
@ynezz 
nothing has changed - everything still applies cleanly an d apart from the cosmetic and subjective changes requested, I am still waiting for wifi drivers to update to at least 5.9 (yes, backports is still taking from 5.8 even though its been months) for the final commit cleaning up name clashes.

**Comment by adschm** [comment]
> @ynezz
> nothing has changed - everything still applies cleanly an d apart from the cosmetic and subjective changes requested, I am still waiting for wifi drivers to update to at least 5.9 (yes, backports is still taking from 5.8 even though its been months) for the final commit cleaning up name clashes.

Well, e.g. `default y if TARGET_imx6_DEVICE_cubox-i` is simply broken (because of wrong assumptions), I don't see where this is subjective or cosmetic.

**Comment by adschm** [comment]
Since requested changes have not been made, I'm closing this. Feel free to request reopening when you have time/interest to continue.

**Comment by Josua-SR** [comment]
> > @ynezz
> > nothing has changed - everything still applies cleanly an d apart from the cosmetic and subjective changes requested, I am still waiting for wifi drivers to update to at least 5.9 (yes, backports is still taking from 5.8 even though its been months) for the final commit cleaning up name clashes.
> 
> Well, e.g. `default y if TARGET_imx6_DEVICE_cubox-i` is simply broken (because of wrong assumptions), I don't see where this is subjective or cosmetic.
> 
> Apart from that, stuff like proper line-wrapping is still a hard requirement.

I can simply assure you that it worked. I built openwrt for a single target (cubox-i), and got expected behaviour.

**Pull request closed without merging**